### ISSUE TITLE
Using the Builder to create a Consul Object, URL is cutted

### ISSUE THREAD

**Issue by Bryann** [OrbitzWorldwide/consul-client#142]
Hi,

i'm currently facing the following issue:

I try to create a new Consul object using the `Consul.builder().withUrl(urlString);`
The urlString has the following pattern: 

`https://<IP>/path/to/consul`

If i build my object using `build()`, the `createRetrofit()` Method is executed.
Within this method, my given URL will be cutted/modified to `https://<IP>/v1` instead of `https://<IP>/path/to/consul/v1` 
This leads to a 404 and the Consul API is not reachable.

Is there a reason for this? I tried to change the Code in Consul.java to the following:

```
            return new Retrofit.Builder()
// new way
            		.baseUrl(consulUrl.toExternalForm() + "/v1/")
// old way
//                    .baseUrl(new URL(consulUrl.getProtocol(), consulUrl.getHost(),
//                            consulUrl.getPort(), "/v1/").toExternalForm())
                    .addConverterFactory(JacksonConverterFactory.create(mapper))
                    .client(builder.build())
                    .build();
```

This works fine for me. But i don't think its a solution for all possible use cases.

**Comment by rickfast** [comment]
I'll take a look at this soon

**Comment by rickfast** [comment]
fixed by a4cfd6aa94ad430c17872069205b427f26a75bbc

**Comment by Bryann** [comment]
Thanks for the fix,

can you create a new release for the it?

**Comment by rickfast** [comment]
this fix is in `0.12.8`
